realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -109,7 +118,11 @@ private RealmCache(RealmConfiguration config) {
             // Create a new local Realm instance
             BaseRealm realm;
 
+
             if (realmClass == Realm.class) {
+                if (configuration.getDurability() != SharedGroup.Durability.MEM_ONLY) {
+                    copyAssetFile(configuration);
```

Comment:
I think line 111 is a better place to call `copyAssetFile(configuration);`. Line 111 will run when the first `Realm`/`DynamicRealm` created across all threads. Here is the first typed `Realm.getInstance()` called in the current thread.